src/main.rs

```diff
@@ -83,7 +91,15 @@ fn main() -> anyhow::Result<()> {
         unsupported => panic!("unsupported language: {}", unsupported),
     };
 
-    ir.synthesize(synthesizer.as_ref(), &mut output)?;
+    let stack_name: &str = match matches
+        .get_one::<String>("stack-name")
+        .map(String::as_str)
+        .unwrap_or("NoctStack") 
+    {
+        stack_name => stack_name,
+    };
```

Comment:
Most of the other options had the `unwrap_or` as essentially "if null", e.g.
```
    let mut output: Box<dyn io::Write> = match matches
        .get_one::<String>("OUTPUT")
        .map(String::as_str)
        .unwrap_or("-")
    {
        "-" => Box::new(io::stdout()),
        output_file => Box::new(fs::File::create(output_file)?),
    };
```

Since your code space is always doing the same action, you can simplify it to have no matches, so:
```
    let stack_name = matches
        .get_one::<String>("stack-name")
        .map(String::as_str)
        .unwrap_or("NoctStack");
```

(I also remove the `: &str`, as it was unambiguously determined on the right side)